### ISSUE TITLE
What can I / should I expect when working with huge files (100 MB+)?

### ISSUE THREAD

**Issue by fabiorzfreitas** [psf/black#4127]
This is a purely informative question, any issue I had with this I already managed to solve by other means.

The starting point is this: Windows 11 (in a semi-bloated environment), not a great laptop (8GB RAM and i7, but still) and everything running on an SSD (NVMe, PCIe 2x4, IIRC). And the huge file I attempted to pass to *black*:

```
Before formatting

  1 676 172 Lines
242 696 884 Characters
        219 MB


After formatting

  4 802 985 Lines
290 934 893 Characters
        277 MB
```

This was a non-python output of hashing over a million files (font files grouped as the largest collection on the internet) for duplicates.

With a few RegExes, my starting file to parse looked like this:

``` python

duplicate_groups = [

[
("filename", "path"),
("filename2", "path"),
("filename3", "path3")
],
[
"..."
]
]

```

All I needed/wanted was the indentation and since VSCode reject the file by its size, I was curious about other tools:

```python
duplicate_groups = [
    [
        (
            ".",
            ".."
        ),
        (
            "~",
            "~~"
        )
    ]
]
```

---

## **tl;dr**

I passed a 220 MB file to *black* both as a script and as a module and I let both instances running overnight, neither could do the job.

Another few RegExes and I got my indentation, whatever.

As expected, but I wonder if someone else experimented with this...

**Comment by hauntsaninja** [comment]
Thanks for the issue! I think the use case is outside of normal operating conditions, but it's nice for things to work. If you provide a script that makes a synthetic repro of your case, maybe some volunteer will figure out optimisations that make Black work well against it.
